### PR TITLE
Support nest virt on host

### PR DESCRIPTION
**Pull request by dzhengfy**
Depend on https://github.com/avocado-framework/avocado-vt/pull/3127
Signed-off-by: Dan Zheng <dzheng@redhat.com>

### CODE REVIEW

File: virttest/utils_libvirt/libvirt_nested.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    libvirtd.restart()
+
+
+def enable_nested_virt_on_host():
+    """
+    Enable nested virt on the host
+
+    :raises: exceptions.TestError if nested virt is not set successfully
+    """
+    cmd = 'modprobe -r kvm_intel; modprobe -r kvm; modprobe kvm nested=1; modprobe kvm_intel nested=1'
```
Comment (Yingshun) [inline virttest/utils_libvirt/libvirt_nested.py:33]: Do we need to consider AMD as well?

**Review by kamilvarga — APPROVED** [review]
LGTM.

File: virttest/utils_libvirt/libvirt_nested.py (hunk 1 above)
Comment (dzhengfy) [reply virttest/utils_libvirt/libvirt_nested.py:38]: @Yingshun , confirmed with feature owner, only Intel is needed to be considered. I updated.

File: virttest/utils_libvirt/libvirt_nested.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    :param vm_session: session to the vm
+    :raises: exceptions.TestError if installation fails
+    """
+    pkg_names = ['libvirt', 'qemu-kvm']
+    logging.debug("Virt packages will be installed")
+    pkg_mgr = utils_package.package_manager(vm_session, pkg_names)
+    if not pkg_mgr.install():
+        raise exceptions.TestError("Package '%s' installation "
+                                   "fails" % pkg_names)
+    libvirtd = utils_libvirtd.Libvirtd()
```
Comment (Yingshun) [inline virttest/utils_libvirt/libvirt_nested.py:24]: I don't quite understand why you restart libvirtd on host, would you mind adding some explanations?  libvirt* packages are installed on VM, so do you need to restart libvirtd service on the VM?

Comment (dzhengfy) [reply virttest/utils_libvirt/libvirt_nested.py:24]: Yes, you are right. That's not my original purpose. I will update it.

Comment (dzhengfy) [reply virttest/utils_libvirt/libvirt_nested.py]: updated @Yingshun

**Review by Yingshun — APPROVED** [review]

File: virttest/utils_libvirt/libvirt_nested.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+def install_virt_pkgs(vm_session):
+    """
+    Install required virt packages for nested virt test
+
+    :param vm_session: session to the vm
+    :raises: exceptions.TestError if installation fails
+    """
+    pkg_names = ['libvirt', 'qemu-kvm']
+    logging.debug("Virt packages will be installed")
```
Comment (kylazhang) [inline virttest/utils_libvirt/libvirt_nested.py:19]: I think here should be logging.info?

File: virttest/utils_libvirt/libvirt_nested.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+    cmd = 'cat /sys/module/kvm_intel/parameters/nested'
+    ret = process.run(cmd, verbose=True,
+                      ignore_status=False,
+                      shell=True).stdout_text.strip()
+    if ret != 'Y':
+        raise exceptions.TestError("Checking nested virt environment "
+                                   "with the command '{}' fails with "
+                                   "the result '{}'".format(cmd,
+                                                            ret))
```
Comment (kylazhang) [inline virttest/utils_libvirt/libvirt_nested.py:49]: why leave 'ret))' in the new line?

File: virttest/utils_libvirt/libvirt_nested.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    Update vm cpu xml for nested virt environment
+
+    :param guest_xml: the vm xml
+    :param cpu_mode: like 'host-model', 'host-passthrough'
+    :return: the updated vm xml
+    """
+    # Update cpu mode if needed
+    cur_vmcpuxml = guest_xml.cpu
+    if cpu_mode:
+        logging.debug("Update cpu mode from '{}' to '{}'".format(cur_vmcpuxml.mode, cpu_mode))
```
Comment (kylazhang) [inline virttest/utils_libvirt/libvirt_nested.py:63]: As above, suggest use logging.info

**Review by kylazhang — COMMENTED** [review]
Others are OK for me

File: virttest/utils_libvirt/libvirt_nested.py (hunk 3 above)
Comment (dzhengfy) [reply virttest/utils_libvirt/libvirt_nested.py:19]: updated

File: virttest/utils_libvirt/libvirt_nested.py (hunk 4 above)
Comment (dzhengfy) [reply virttest/utils_libvirt/libvirt_nested.py:49]: updated

File: virttest/utils_libvirt/libvirt_nested.py (hunk 5 above)
Comment (dzhengfy) [reply virttest/utils_libvirt/libvirt_nested.py:63]: updated

**Review by kylazhang — APPROVED** [review]

**Comment by kylazhang** [comment]
@dzhengfy  please check the spellcheck with the error and try fix them before merge. Thanks

**Comment by chloerh** [comment]
Guys, I have submitted pr to update the spell.ignore list:

https://github.com/avocado-framework/avocado-vt/pull/3198

**Comment by dzhengfy** [comment]
> Guys, I have submitted pr to update the spell.ignore list:
> 
> #3198

THanks, Chloerh

**Pull request merged**